NuRadioReco/utilities/trace_utilities.py

```diff
@@ -111,3 +112,116 @@ def get_electric_field_energy_fluence(electric_field_trace, times, signal_window
+        Upsampling factor. The resulting trace will have a sampling frequency
+        int_factor times higher than the original one
+    ntaps: integer
+        Number of taps (order) of the FIR filter
+
+    Returns
+    -------
+    upsampled_trace: array of floats
+        The upsampled trace
+    """
+
+    if (np.abs(int(int_factor) - int_factor) > 1e-3):
+        warning_msg  = "The input upsampling factor does not seem to be close to an integer."
+        warning_msg += "It has been rounded to {}".format(int(int_factor))
+        logger.warning(warning_msg)
+
+    int_factor = int(int_factor)
+
+    if (int_factor <= 1):
+        error_msg = "Upsampling factor is less or equal to 1. Upsampling will not be performed."
+        raise ValueError(error_msg)
+
+    n_zeroes = int_factor - 1
+    zeroed_trace = []
+    for point in trace[:-1]:
+        zeroed_trace.append(point)
+        for zero in range(n_zeroes):
+            zeroed_trace.append(0.)
+
+    upsampled_delta_time = 1 / (int_factor * original_sampling_frequency)
+    upsampled_times = np.arange(0, len(zeroed_trace) * upsampled_delta_time, upsampled_delta_time)
+
+    cutoff = 1./int_factor
+    fir_coeffs = firwin(ntaps, cutoff)
+    upsampled_trace = np.convolve(zeroed_trace, fir_coeffs)[:len(upsampled_times)] * int_factor
+
+    return upsampled_trace
+
+def butterworth_filter_trace(trace, sampling_frequency, passband, order=8):
+    """
+    Filters a trace using a Butterworth filter.
+
+    Parameters
+    ----------
+    trace: array of floats
+        Trace to be filtered
+    sampling_frequency: float
+        Sampling frequency
+    passband: (float, float) tuple
+        Tuple indicating the cutoff frequencies
+    order: integer
+        Filter order
+
+    Returns
+    ------
+    filtered_trace: array of floats
+        The filtered trace
+    """
+
+    n_samples = len(trace)
+
+    spectrum = fft.time2freq(trace, sampling_frequency)
+    frequencies = np.fft.rfftfreq(n_samples, 1/sampling_frequency)
+
+    filtered_spectrum = apply_butterworth(spectrum, frequencies, passband)
+    filtered_trace = fft.freq2time(filtered_spectrum, sampling_frequency)
+
+    return filtered_trace
+
+def apply_butterworth(spectrum, frequencies, passband, order=8):
+    """
+    Calculates the response from a Butterworth filter and applies it to the
+    input spectrum
+
+    Parameters
+    ----------
+    spectrum: array of complex
+        Fourier spectrum to be filtere
+    frequencies: array of floats
+        Frequencies of the input spectrum
+    passband: (float, float) tuple
+        Tuple indicating the cutoff frequencies
+    order: integer
+        Filter order
+
+    Returns
+    -------
+    filtered_spectrum: array of complex
+        The filtered spectrum
+    """
+
+    f = np.zeros_like(frequencies, dtype=np.complex)
+    mask = frequencies > 0
+    b, a = butter(order, passband, 'bandpass', analog=True)
+    w, h = freqs(b, a, frequencies[mask])
```

Comment:
can you change the imports to `from scipy import signal.freqs` such that you need to call this function via `signal.freqs` or even better via `scipy.signal.freqs`. `freqs` is common variable name that will just override the function. So leaving it like this can be error prone.

Reply:
Changed.